v2/pkg/engine/resolve/load.go

```diff
@@ -451,7 +462,54 @@ func (l *Loader) resolveFetch(ctx *Context, fetch Fetch, res *resultSet) (err er
 	return nil
 }
 
-func (l *Loader) resolveBatchFetch(ctx *Context, fetch *BatchFetch, res *resultSet) (err error) {
+func (l *Loader) resolveEntityFetch(ctx *Context, fetch *EntityFetch, res *resultSet) (err error) {
+	input := res.getBuffer()
+	out := res.getBuffer()
+
+	lr := l.currentLayer()
+	err = fetch.Input.Header.Render(ctx, nil, input)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	itemBuf := pool.BytesBuffer.Get()
+	defer pool.BytesBuffer.Put(itemBuf)
+
+	err = fetch.Input.Item.Render(ctx, lr.data, itemBuf)
+	if err != nil {
+		if fetch.Input.SkipErrItem {
+			err = nil
+			// skip fetch on render item error
+			return
+		}
+		return errors.WithStack(err)
+	}
+
+	switch {
+	case itemBuf.Len() == 4 && bytes.Equal(itemBuf.Bytes(), null):
+		// skip fetch if item is null
+		return
+	case itemBuf.Len() == 2 && bytes.Equal(itemBuf.Bytes(), emptyObject):
+		// skip fetch if item is empty
+		return
+	}
+
+	_, _ = input.Write(itemBuf.Bytes())
+
+	err = fetch.Input.Footer.Render(ctx, nil, input)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	res.mergePath = fetch.PostProcessing.MergePath
```

Comment:
Is this the right place to set mergePath?
I guess yes, just asking.
Seems better to abstract it away somehow.